improve and test vite-plugin-cloudflare `.dev.vars` files support (#7864)
cloudflare/workers-sdk: .changeset/orange-camels-hunt_vite-plugin.md

---
"@cloudflare/vite-plugin": patch
---

> add full support for `.dev.vars` files
> 
> this change makes sure that `.dev.vars` files work for a specified environment, it also
> copies the target `.dev.vars` file (which might be environment specific, eg: `.dev.vars.prod`)
> to the worker's dist directory, always named `.dev.vars` (so that `vite preview` can pick it up)

Reviewer: ```suggestion
Add full support for `.dev.vars` files.

This change makes sure that `.dev.vars` files work when the environment is specified. It also
copies the target `.dev.vars` file (which might be environment specific, e.g. `.dev.vars.prod`)
to the worker's dist directory so that `vite preview` can pick it up. The copied file will always be named `.dev.vars`.
```
